budgetportal/models.py

```diff
@@ -1412,6 +1416,148 @@ class PackageWithoutGroupException(Exception):
     pass
 
 
+class InfrastructureProject():
+    """ Represents an infrastructure project stored in CKAN """
+
+    def __init__(self, **kwargs):
+        self.records = kwargs.get('records')
+        if self.records:
```

Comment:
when would you want an InfrastructureProject instance that isn't initialised with data? I think you'd normally want to be able to assume it's initialised or instantiation should fail